### ISSUE TITLE
Update test suite for exercise: phone-number

### ISSUE THREAD

**Issue by ErikSchierboom** [exercism/fsharp#590]
The `phone-number` exercise's [test suite](https://github.com/exercism/fsharp/blob/master/exercises/phone-number/PhoneNumberTest.fs) is automatically generated using its [test generator](https://github.com/exercism/fsharp/blob/master/generators/Generators.fs). This test generator converts the `phone-number`'s [canonical data](https://github.com/exercism/problem-specifications/blob/master/exercises/phone-number/canonical-data.json) into the test suite just mentioned.

The canonical data for the `phone-number` exercise has been updated since the test suite was last generated. We should be updating thetest suite to the latest version. To do this, one should:
- Run the test generator for the exercise. This is done by opening a command prompt in the `generators` directory and running: `dotnet run -e phone-number`.
- At this point, you should check to see what the updated test suite looks like. There are basically two options: one, the test generator was compatible with the updated canonical data and the test suite was updated and still has the same structure. The second option is that the test generator could not process the changed canonical data, in which case the generator has to be modified. In that case, you should check out the [generator docs](https://github.com/exercism/fsharp/blob/master/docs/GENERATORS.md) in which the generators setup is explained in detail.
- Finally, please make sure that the [example implementation](https://github.com/exercism/fsharp/blob/master/exercises/phone-number/Example.fs) passes all the tests. You can verify this by opening a command prompt in the root directory and running `./build.sh --exercise=phone-number` or `.\build.ps1 -Exercise phone-number`, depending on your operating system.

**Comment by ErikSchierboom** [comment]
This is odd. I'll get back to you tomorrow

**Comment by ErikSchierboom** [comment]
I've looked into this a bit, and it is really weird. If I run `dotnet run -e phone-number`, the test file gets updated. This works both on my two machines at home (Windows and Mac), and my work machine. Let's see if we can figure out what goes wrong.

First, could you print the console output that you get when running the command?

**Comment by ErikSchierboom** [comment]
Note: the problem has been located! The instructions forgot to mention that you have to run the dotnet command in the `generators` directory. Sorry about that.

**Comment by MaritimeMartin** [comment]
For the moment that are the changes in Generator.fs:

```
override __.RenderExpected (_, _, value) =
        match value.SelectToken "error" with
        | null  -> 
            value
            |> string
            |> sprintf "Ok %sUL"
        | error -> 
            error
            |> string
            |> sprintf "Error \"%s\""
        |> String.parenthesize

```

Was before:

```
override __.RenderExpected (_, _, value) =
        value 
        |> Option.ofNonNull
        |> Option.renderParenthesized 
```

Before the new canonical data, there was no further error information, just `Some BigInt` or `None` . Now the error messages are quite granular so that the Example.fs need to be altered. I am on it but not sure about my concept for that file.

**Comment by ErikSchierboom** [comment]
> I am on it but not sure about my concept for that file.

You should indeed modify the Example.fs file to match the updated test suite. 

Coudl you perhaps also bump the difficulty of this exercise one point in the `config.json` file?

**Comment by MaritimeMartin** [comment]
I would like to have a valid solution for this issue first because I can't estimate the difficulty before you had an eye on the solution and told me, that it is really this difficult.

**Comment by ErikSchierboom** [comment]
Closed by #637
